packages/neo-one-smart-contract-lib/src/ownership/Ownable.ts

```diff
@@ -0,0 +1,52 @@
+import { Address, constant, createEventNotifier, SmartContract } from '@neo-one/smart-contract';
+
+/**
+ * @title Ownable
+ * @dev A ownable contract has an owner address
+ */
+
+/* tslint:disable-next-line: variable-name */
+const transferred_ownership = createEventNotifier<Address | undefined, Address>('transferred ownership', 'from', 'to');
+
+export function Ownable<TBase extends Constructor<SmartContract>>(base: TBase) {
+  class OwnableClass extends base {
+    protected mutableOwner: Address | undefined;
+    public constructor(...args: any[]) {
+      super();
+      this.mutableOwner = args[0];
+    }
+
+    @constant
+    public get owner(): Address | undefined {
+      return this.mutableOwner !== undefined ? this.mutableOwner : undefined;
+    }
+
+    @constant
+    public onlyOwner() {
+      if (this.mutableOwner && !Address.isCaller(this.mutableOwner)) {
+        throw new Error('not owner');
+      }
+    }
+
+    public protected(to: Address): boolean {
+      return this._transferOwnership(to);
+    }
+    public transferOwnership(to: Address): boolean {
+      return this._transferOwnership(to);
+    }
+    public renounceOwnership() {
+      this.onlyOwner();
+
+      this.mutableOwner = undefined;
+    }
+    protected _transferOwnership(to: Address): boolean {
+      this.onlyOwner();
```

Comment:
The implementation here can just be moved to `transferOwnership`

Reply:
❓ Again following the logic division as it is demonstrated in the templates we were following. 
With that said: Leave or Move?  
I did move `onlyOwner()` check tho.

Reply:
Move - we don't need to translate it identically, in fact we should not translate identically in the cases where it's not idiomatic typescript.

Reply:
👍